dace/data.py

```diff
@@ -342,6 +341,151 @@ def add(X: dace.float32[10, 10] @ dace.StorageType.GPU_Global):
         return new_desc
 
 
+def _arrays_to_json(arrays):
+    if arrays is None:
+        return None
+    return {k: serialize.to_json(v) for k, v in arrays.items()}
+
+
+def _arrays_from_json(obj, context=None):
+    if obj is None:
+        return {}
+    return {k: serialize.from_json(v, context) for k, v in obj.items()}
+
+
+@make_properties
+class Structure(Data):
+    """ Base class for structures. """
+
+    members = Property(dtype=dict,
```

Comment:
This needs to be ordered. I'd use a list property of tuples

Reply:
Does it? `members` is just a registry of the nested data for use with the SDFG API. My understanding is that the order of the `members` in memory depends exclusively on the Structure's `dtype`, which is a (pointer to) `dtypes.struct`. As long as this is (de)serialized consistently, it should be OK. However, looking at the code, it seems that no special care is taken in `dtypes.struct` to enforce a specific ordering. Should we update the code in `dtypes.struct` to order the struct's fields alphabetically or something appropriate?

Reply:
To properly marshal external data, I’d just order the property both in dace.struct and here